### PR TITLE
busybox: fix compatibility with BUSYBOX_CONFIG_INSTALL_NO_USR

### PR DESCRIPTION
**Pull request by ghost**
Fixes two bugs that manifest when  BUSYBOX_CONFIG_INSTALL_NO_USR is selected.
See each commit for details.

Signed-off-by: Marius Dinu <m95d+git@psihoexpert.ro>

### CODE REVIEW

**Comment by M95D** [comment]
Why is busybox installed in /usr? It creates problems like [overwriting full programs with bb applets when updating busybox](https://forum.openwrt.org/t/limit-busybox-applets-to-bin-and-sbin/71915). Isn't /usr for user programs on all unix systems? Busybox is a core package. It dosn't belong there.
New option:
* Move busybox by default to /bin and /sbin and make BUSYBOX_CONFIG_INSTALL_NO_USR permanently enabled.

**Comment by M95D** [comment]
When busybox is built with BUSYBOX_CONFIG_INSTALL_NO_USR, /usr/bin/env applet is created in /bin/env instead. This breaks Luci.
Added a separate fix that creates the symlink as /usr/bin/env.

**Comment by M95D** [comment]
I think this is ready for commit. I found no other incompatibilities.

**Comment by M95D** [comment]
@jow-  @hauke 
Please review.

File: package/utils/busybox/Makefile [hunk 1]
```diff
@@ -155,7 +163,7 @@ ifneq ($(CONFIG_BUSYBOX_$(BUSYBOX_SYM)_NTPD),)
 	$(INSTALL_DATA) ./files/ntpd.capabilities $(1)/etc/capabilities/ntpd.json
 	$(INSTALL_DATA) ./files/ntpd_acl.json $(1)/usr/share/acl.d/ntpd.json
 endif
-	-rm -rf $(1)/lib64
```
Comment (hauke) [inline package/utils/busybox/Makefile:158]: Why is this remove not needed any more?

File: package/utils/busybox/Makefile [hunk 2]
```diff
@@ -138,8 +138,16 @@ define Build/Compile
 endef
 
 define Package/busybox/install
-	$(INSTALL_DIR) $(1)/etc/init.d
+
+ifneq ($(CONFIG_BUSYBOX_$(BUSYBOX_SYM)_INSTALL_NO_USR),)
+	mkdir -p $(PKG_INSTALL_DIR)/usr/bin
```
Comment (hauke) [inline package/utils/busybox/Makefile:143]: Why don't you use `$(INSTALL_DIR) $(1)/usr/bin` ?

File: package/utils/busybox/Makefile [hunk 3]
```diff
@@ -138,8 +138,16 @@ define Build/Compile
...
+	ln -s -r -f $(PKG_INSTALL_DIR)/bin/busybox $(PKG_INSTALL_DIR)/usr/bin/env
+	$(INSTALL_DIR) $(1)/bin
+	$(INSTALL_DIR) $(1)/sbin
+endif
+	$(INSTALL_DIR) $(1)/usr/bin
```
Comment (hauke) [inline package/utils/busybox/Makefile:148]: Why is `$(INSTALL_DIR) $(1)/usr/bin` added here?

**Comment by M95D** [comment]
@hauke Polite ping.

File: package/utils/busybox/files/sysntpd [hunk 4]
```diff
@@ -4,7 +4,7 @@
 START=98
 
 USE_PROCD=1
-PROG=/usr/sbin/ntpd
+PROG=ntpd
```
Comment (hauke) [inline package/utils/busybox/files/sysntpd:7]: jail does not allow this.
procd fails like this:
````
Sun Mar  5 23:08:39 2023 user.err : jail: ntpd is not an absolute path
````

Comment (M95D) [reply package/utils/busybox/files/sysntpd:7]: How about this?
PROG=$(which ntpd)

If not, BUSYBOX_CONFIG_INSTALL_NO_USR should be permanently removed.

**Pull request closed without merging**